doc/rst/developer/bestPractices/ContributorInfo.rst

```diff
@@ -9,80 +9,297 @@ The git repository for the project is hosted on Github at
-    code. Reviewers on the core team can pull, review, and merge your pull
-    requests.
+Set up repository
++++++++++++++++++
 
-* If you're working on a long-term effort, announce it on the
-  chapel-developers_ to make sure toes are not being stepped on, work is not
-  being pursued redundantly, etc.  Similarly, fundamental changes to the
-  language or architecture should be circulated to the chapel-developers_
-  and/or chapel-users_ lists to make sure effort is not wasted.
+Note: these are expected to evolve over time as the governance of Chapel is
+migrated from Cray to an external/community body (the major elements are likely
+to be similar, though the specific people involved are likely to change and
+grow).
 
-* If your work will require committing any third-party code that you are not
-  developing yourself (or code that you've developed as a standalone package),
-  alert the chapel-developers_ mailing list of this as, presently, such code
-  packages must be approved by Cray leadership before being committed.
+#. `Set up a github account`_. The "Free" plan is sufficient for contributing to
+   Chapel.
 
-* Sign a Chapel contributor's agreement and mail it, with your GitHub
-  ID, using the instructions here:
+#. Make sure you have configured your environment to work with git. See
+   `initial git setup`_ instructions.
 
-  https://github.com/chapel-lang/chapel/tree/master/doc/rst/developer/contributorAgreements/
+#. Use the GitHub web interface to create a fork of the Chapel repo by visiting
+   https://github.com/chapel-lang/chapel and clicking the 'Fork' button (see
+   also `Fork the repo`_).  Then `configure your local git`_ and check out your
+   fork
 
-* `Fork the repo`_.
+#. If you will need commit/push access to the main repository,
+   `chapel-lang/chapel`_, send a request including your github username to
+   chapel_admin _at_ cray.com.
+
+   Note: You do not need commit/push access to the main repo in order to
+   contribute code. Reviewers on the core team can pull, review, and merge your
+   pull requests.
+
+#. If you're working on a long-term effort, announce it on the
+   chapel-developers_ mailing list to make sure toes are not being stepped on,
+   work is not being pursued redundantly, etc.  Similarly, fundamental changes
+   to the language or architecture should be circulated to the
+   chapel-developers_ and/or chapel-users_ lists to make sure effort is not
+   wasted.
+
+#. If your work will require committing any third-party code that you are not
+   developing yourself (or code that you've developed as a standalone package),
+   alert the chapel-developers_ mailing list of this as, presently, such code
+   packages must be approved by Cray leadership before being committed.
+
+#. Sign a Chapel contributor's agreement and mail it, with your GitHub
+   ID, using the instructions here:
+
+   https://github.com/chapel-lang/chapel/tree/master/doc/rst/developer/contributorAgreements/
```

Comment:
Is this the right place to bring up the contributor agreement? We could also request it at the time the PR is created, which is where we check in practice. Here, it might suffice to say something like "You will need to sign a contributor agreement for your changes to be accepted" and then refer to the link.